packaging/common/rabbitmq-server-ha.ocf

```diff
@@ -0,0 +1,1585 @@
+    for i in $(iptables -nvL --line-numbers | awk '/temporary RMQ block/ {print $1}'); do
+      iptables -D INPUT -p tcp -m tcp --dport ${OCF_RESKEY_node_port} -m state --state NEW,RELATED,ESTABLISHED \
+      -m comment --comment 'temporary RMQ block' -j REJECT --reject-with tcp-reset
+    done
+}
+
+get_nodes__base(){
+    local infotype=''
+    local rc=$OCF_ERR_GENERIC
+
+    if [ "$1" == 'nodes' ]
+    then
+        infotype='db_nodes'
+    elif [ "$1" == 'running' ]
+    then
+        infotype='running_db_nodes'
+    fi
+    local c_status=$(${OCF_RESKEY_ctl} eval "mnesia:system_info(${infotype})." 2>/dev/null)
+    rc=$?
+    if [[ $rc != 0 ]] ; then
+        echo ''
+        return $OCF_ERR_GENERIC
+    fi
+    # translate line like '{running_nodes,['rabbit@node-1','rabbit@node-2','rabbit@node-3']},' to node_list
+    echo $(echo "${c_status}" | grep "${cl}" | awk -F, '{ for (i=1;i<=NF;i++) { if ($i ~ /@/) { gsub(/[\[\]}{]/,"",$i); print $i; } }}' | tr -d  "\'")
+    return $OCF_SUCCESS
+}
+
+get_nodes() {
+    echo $(get_nodes__base nodes)
+    return $?
+}
+
+get_running_nodes() {
+    echo $(get_nodes__base running)
+    return $?
+}
+
+# Get all known cluster nodes including offline ones
+get_all_pacemaker_nodes()
+{
+    echo `crm_node -l | awk '{print $2}' | grep -v "^$" | sed -e '/(null)/d'`
+    return $?
+}
+
+# Get alive cluster nodes in visible partition, but the specified one
+get_alive_pacemaker_nodes_but()
+{
+    if [ -z $1 ]; then
+        echo `crm_node -l -p | sed -e '/(null)/d'`
+    else
+        echo `crm_node -l -p | sed -e "s/${1}//g" | sed -e '/(null)/d'`
+    fi
+    return $?
+}
+
+check_need_join_to() {
+    local join_to=$(rabbit_node_name $1)
+    local node
+    local running_nodes=$(get_running_nodes)
+    local rc=$OCF_ERR_GENERIC
+
+    rc=0
+    for node in $running_nodes ; do
+        if [[ ${join_to} == ${node} ]] ; then
+            rc=1
+            break
+        fi
+    done
+
+    return $rc
+}
+
+# Update erlang cookie, if it has been specified
+update_cookie() {
+    if [[ "${OCF_RESKEY_erlang_cookie}" != false ]] ; then
+        echo "${OCF_RESKEY_erlang_cookie}" > "${OCF_RESKEY_erlang_cookie_file}" && \
+        chown ${OCF_RESKEY_username}:${OCF_RESKEY_groupname} "${OCF_RESKEY_erlang_cookie_file}" && \
+        chmod 600 "${OCF_RESKEY_erlang_cookie_file}"
+    fi
+    return $OCF_SUCCESS
+}
+
+kill_rmq_and_remove_pid() {
+    local pid
+    local LH="${LL} kill_rmq_and_remove_pid():"
+
+    if [[ -f $OCF_RESKEY_pid_file ]] ; then
+        pid=$(cat $OCF_RESKEY_pid_file)
+        if [[ -z ${pid} ]] ; then
+            ocf_log err "${LH} pidfile is empty, cannot kill by unknown PID! Try to stop it manually!"
+        fi
+        # todo: check content for digital
+        if [[ -d /proc/${pid}/ ]] ; then
+            ocf_run kill -9 $pid
+            ocf_log warn "${LH} RMQ-runtime (beam) PID=${pid} stopped by 'kill -9', sorry..."
+        fi
+        ocf_run rm -f $OCF_RESKEY_pid_file
+    fi
```

Comment:
Would it be worth checking for a running `beam` or `beam.smp` without the associated PID file? `beam` command line contains the node name.